ironfish/src/workerPool/pool.ts

```diff
@@ -235,22 +222,16 @@ export class WorkerPool {
     nonce: string,
     sender: Identity,
     recipient: PrivateIdentity,
-  ): Promise<{ message: string | null }> {
-    const request: UnboxMessageRequest = {
-      type: 'unboxMessage',
-      boxedMessage: boxedMessage,
-      nonce: nonce,
-      recipient: recipient,
-      sender: sender,
-    }
+  ): Promise<UnboxMessageResponse> {
+    const request = new UnboxMessageRequest(boxedMessage, nonce, sender, recipient)
 
     const response = await this.execute(request).result()
 
-    if (response === null || response.type !== request.type) {
+    if (!(response instanceof UnboxMessageResponse)) {
       throw new Error('Response type must match request type')
```

Comment:
Do we want to be consistent with the `Invalid response` error return? Or do we want to be more descriptive?